modules/reindex/src/main/java/org/elasticsearch/reindex/remote/RemoteRequestBuilders.java

```diff
@@ -140,11 +141,22 @@ static Request initialSearch(SearchRequest searchRequest, BytesReference query,
                 }
             }
 
-            if (searchRequest.source().fetchSource() != null) {
-                entity.field("_source", searchRequest.source().fetchSource());
-            } else {
-                if (remoteVersion.onOrAfter(Version.fromId(1000099))) {
-                    // Versions before 1.0 don't support `"_source": true` so we have to ask for the source as a stored field.
+            if (remoteVersion.onOrAfter(Version.fromId(1000099))) {
+                // Versions before 1.0 don't support `_source` so we have to ask for the source as a stored field.
+                if (searchRequest.source().fetchSource() != null) {
+                    FetchSourceContext fetchSource = searchRequest.source().fetchSource();
+                    if (remoteVersion.before(Version.V_8_16_0)) {
+                        // Versions before 8.16 do not support the include_vectors param. They include all vector fields by default.
+                        // Set the include_vectors param to the default value so that it is not included in XContent.
+                        fetchSource = FetchSourceContext.of(
+                            fetchSource.fetchSource(),
+                            fetchSource.includes(),
+                            fetchSource.excludes(),
+                            FetchSourceContext.DEFAULT_INCLUDE_VECTORS
+                        );
+                    }
+                    entity.field("_source", fetchSource);
+                } else {
```

Comment:
Had to make two key changes here:

- Set `include_vectors` to the default value when sending a request to a cluster older than 8.16. This prevents the `include_vectors` value from being emitted in XContent.
- Don't add `_source` when sending a request to a cluster older than 1.0. It appears that there was a long-standing bug in this code where if `searchRequest.source().fetchSource() != null`, `_source` would be added regardless of the remote cluster version. This became a problem due to changes in this PR, specifically because we now always create a `FetchSourceContext` when reindexing (see changes to `AbstractBulkIndexByScrollRequest`). Previously, `searchRequest.source().fetchSource()` was almost always null when reindexing, hiding this bug.